File: Launcher/src/models/tool_input.py
"""
Tool Configuration Input Model
"""


class ToolInput:
    """
    Tool Configuration Model
    """

    def __init__(self, key: str, description: str, format_type: str):
        self.key = key
        self.description = description
        self.format_type = format_type

    def __str__(self):

        key = f"key={self.key}"
        desc = f"description={self.description}"
        format_type = f"format_type={self.format_type}"

        return f"ToolInput: {key}, {desc}, {format_type}"

File: Backend/OSINT-Collector/src/main/java/com/unina/osintcollector/model/ToolInput.java
package com.unina.osintcollector.model;

import java.util.List;

public class ToolInput {
    private final Tool tool;
    private final Capability capability;
    private final List<Input> inputs;

    public ToolInput(Tool tool, Capability capability, List<Input> inputs) {
        this.tool = tool;
        this.capability = capability;
        this.inputs = inputs;
    }

    public Tool getTool() {
        return tool;
    }

    public List<Input> getInputs() {
        return inputs;
    }

    public Capability getCapability() {
        return capability;
    }
}
